flax/core/meta.py

```diff
@@ -32,7 +32,7 @@
 from jax.experimental import pjit
 
 
-TAxisMetadata = TypeVar('TAxisMetadata', bound='AxisMetadata')
+TAxisMetadata = Any # TypeVar('TAxisMetadata', bound='AxisMetadata')
```

Comment:
mypy doesn't support bound type-vars?!  is this intentional Any cast?

Reply:
mypy has had lots of trouble dealing with type-vars in general. I've tried to preserve their use, and just define them as Any. That way, we can reintroduce stricter checking later where possible.